### PR TITLE
Adding image_pull test

### PR DESCRIPTION
**Pull request by dry923**
Depends-On: https://github.com/cloud-bulldozer/benchmark-operator/pull/565

### Description

Adding image pull workload which takes an image and times how long it takes skopeo to copy it.

### CODE REVIEW

**Comment by jtaleric** [comment]
@dry923 want to resolve whatever conflicts we have?

**Comment by dry923** [comment]
> @dry923 want to resolve whatever conflicts we have?

resolved

**Comment by dry923** [comment]
CI is failing b/c of 
```
snafu/image_pull_wrapper/ci_test.sh: line 11: tests/test_image_pull.sh: No such file or directory
```

However if you look at the benchmark-operator CI (https://github.com/cloud-bulldozer/benchmark-operator/pull/565#issuecomment-819011526) it's there and it works. It's failing due to an issue with the "Depends-On:" flag

File: snafu/image_pull_wrapper/image_pull_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import os
+import argparse
+from .trigger_image_pull import Trigger_image_pull
+
+
+class image_pull_wrapper:
```
Comment (kurtismullins) [inline snafu/image_pull_wrapper/image_pull_wrapper.py:19]: (nitpick) Not the most Pythonic class name (snake_case vs. PascalCase) but I do see that's what is being used by other tests.

File: snafu/image_pull_wrapper/trigger_image_pull.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+            logger.error(err)
+        return results
+
+    def emit_actions(self):
+        logger.info(
+            "Running image pull test with %d image retries and image: %s" % (self.retries, str(self.image))
+        )
+        logger.info("Test UUID is %s on cluster %s" % (self.uuid, self.cluster_name))
+
+        timestamp = time.strftime("%Y-%m-%dT%H:%M:%S")
```
Comment (kurtismullins) [inline snafu/image_pull_wrapper/trigger_image_pull.py:110]: It might be useful to have a common, custom JSONEncoder for all of the tests wrapped by snafu.

**Review by kurtismullins — APPROVED** [review]
LGTM

**Comment by jtaleric** [comment]
LGTM

File: snafu/image_pull_wrapper/image_pull_wrapper.py (hunk 1 above)
Comment (jtaleric) [reply snafu/image_pull_wrapper/image_pull_wrapper.py:19]: yeah, lots of class names can be updated... if we set a standard we should stick to it.

**Pull request merged**